### PR TITLE
Develop/fixes/td 853 course appears twice in the list after successfully enrolled

### PR DESCRIPTION
**Pull request by manishagarwalnhs**
### JIRA link
https://hee-tis.atlassian.net/browse/TD-853

### Description
Enrol Delegate on Activity -  Course appears twice in the list after successfully enrolled 

### Screenshots
https://hee-tis.atlassian.net/browse/TD-853

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [ ] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

File: DigitalLearningSolutions.Data/DataServices/SupervisorDelegateDataService.cs [hunk 1]
```diff
@@ -69,6 +69,7 @@ IEnumerable<string> emails
                         sd.ID,
                         sd.SupervisorAdminID,
                         sd.SupervisorEmail,
+                        sd.CandidateID_deprecated,
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Data/DataServices/SupervisorDelegateDataService.cs:72]: This doesn't appear to relate to this fix and is reinstating a deprecated field, can we revert this change?

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
See query re deprecated field in SQL.

File: DigitalLearningSolutions.Data/DataServices/SupervisorDelegateDataService.cs (hunk 1 above)
Comment (manishagarwalnhs) [reply DigitalLearningSolutions.Data/DataServices/SupervisorDelegateDataService.cs]: change reverted

**Review by martinfoster-hee — COMMENTED** [review]
You can avoid inadvertant changes to your sql connection strings by enabling TCP/IP and named pipes in SQL Server (in windows go to Sql Server Configuration Manager => SQL Server Network Configuration => Protocols for MSSQL Server and set the Named Pipes and TCP/IP settings to Enabled). That way the code with be able to find the sql localhost without having to change the default connection strings.

File: DigitalLearningSolutions.Data/DataServices/SupervisorDelegateDataService.cs [hunk 2]
```diff
@@ -79,7 +80,7 @@ IEnumerable<string> emails
                     INNER JOIN AdminUsers au ON sd.SupervisorAdminID = au.AdminID
                     WHERE au.CentreID = @centreId
                       AND sd.DelegateEmail IN @emails
-                      AND sd.DelegateUserID IS NULL
+                      AND sd.CandidateID_deprecated IS NULL
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Data/DataServices/SupervisorDelegateDataService.cs:83]: This change needs reverting. It doesn't appear to relate to this ticket and is replacing new code with old.

File: DigitalLearningSolutions.Web/appsettings.Development.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "ConnectionStrings": {
-    "DefaultConnection": "Data Source=localhost;Initial Catalog=mbdbx101_uar;Integrated Security=True;"
+    "DefaultConnection": "Data Source=localhost\\MSSQLSERVER01;Initial Catalog=mbdbx101_uar;Integrated Security=True;",
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/appsettings.Development.json:3]: Please don't commit changes to appsettings files unless purposeful.

File: DigitalLearningSolutions.Web/appsettings.json [hunk 4]
```diff
@@ -1,8 +1,8 @@
 {
   "AllowedHosts": "*",
   "ConnectionStrings": {
-    "DefaultConnection": "Data Source=localhost;Initial Catalog=mbdbx101_uar;Integrated Security=True;",
-    "UnitTestConnection": "Data Source=localhost;Initial Catalog=mbdbx101_uar_test;Integrated Security=True;"
+    "DefaultConnection": "Data Source=localhost\\MSSQLSERVER01;Initial Catalog=mbdbx101_uar;Integrated Security=True;",
```
Comment (kevwhitt-hee) [inline DigitalLearningSolutions.Web/appsettings.json:4]: Again, these changes need reverting.

**Review by kevwhitt-hee — CHANGES_REQUESTED** [review]
@manishagarwalnhs, there are several changes in this PR that don't look like they are related to the ticket. Please can they be reverted?

**Comment by kevwhitt-hee** [comment]
@manishagarwalnhs, please take a look at the conflicts on this PR branch and resolve.
